chain/client/src/debug.rs

```diff
@@ -195,6 +203,70 @@ impl Handler<DebugStatus> for ClientActorInner {
     }
 }
 
+fn get_block_hashes_to_fetch(
+    chain_store: &ChainStoreAdapter,
+    height_to_fetch: BlockHeight,
+    final_height: BlockHeight,
+) -> Vec<CryptoHash> {
+    if height_to_fetch >= final_height {
+        chain_store
+            .get_all_header_hashes_by_height(height_to_fetch)
+            .unwrap_or_default()
+            .into_iter()
+            .collect()
+    } else {
+        chain_store.get_block_hash_by_height(height_to_fetch).ok().into_iter().collect()
+    }
+}
+
+fn find_first_height_to_fetch(
+    chain_store: &ChainStoreAdapter,
+    mut height_to_fetch: BlockHeight,
+    mode: DebugBlocksStartingMode,
+    final_height: BlockHeight,
+) -> Result<BlockHeight, near_chain_primitives::Error> {
+    if matches!(mode, DebugBlocksStartingMode::All) {
+        return Ok(height_to_fetch);
+    }
+
+    let min_height_to_search = max(
+        height_to_fetch as i64 - DEBUG_MAX_BLOCKS_TO_SEARCH as i64,
+        chain_store.genesis_height() as i64,
+    ) as u64;
+    while height_to_fetch > min_height_to_search {
+        let block_hashes = get_block_hashes_to_fetch(chain_store, height_to_fetch, final_height);
+        if block_hashes.is_empty() {
+            if matches!(
+                mode,
+                DebugBlocksStartingMode::JumpToBlockMiss | DebugBlocksStartingMode::JumpToChunkMiss
```

Comment:
I guess it's up to you, but I would prob expect `JumpToChunkMiss` to show me the first existing block w some missing chunk in it